infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/environment/convert/DockerImageEnvironmentConverter.java

```diff
@@ -39,31 +40,49 @@
   static final String POD_NAME = "dockerimage";
   static final String CONTAINER_NAME = "container";
 
+  private EntryPointParser entryPointParser = new EntryPointParser();
+
   public KubernetesEnvironment convert(DockerImageEnvironment environment)
       throws InfrastructureException {
-    final Iterator<String> iterator = environment.getMachines().keySet().iterator();
-    if (!iterator.hasNext()) {
-      throw new InternalInfrastructureException(
-          "DockerImage environment must contain at least one machine configuration");
+
+    if (environment.getMachines().size() > 1) {
+      throw new InfrastructureException(
+          format(
+              "Environment of type '%s' doesn't support multiple machines, but contains machines: %s",
+              environment.getType(), String.join(", ", environment.getMachines().keySet())));
+    }
+
+    InternalMachineConfig machine = null;
+    String machineName = POD_NAME;
+
+    if (!environment.getMachines().isEmpty()) {
```

Comment:
Maybe it would be even simpler if `DockerImageEnvironmentFactory` provisions empty machine config if it is not present?
It could fix one small bug: machine memory attribute is not propagated if `dockerimage`  recipe does not have machine config specified. (I can be wrong but it's what I see from code 
https://github.com/eclipse/che/blob/aaa8f424cd198e68e128bca939d9ac058b9ea52e/infrastructures/docker/environment/src/main/java/org/eclipse/che/workspace/infrastructure/docker/environment/dockerimage/DockerImageEnvironmentFactory.java#L85-L89